app/register/register.go

```diff
@@ -95,11 +96,11 @@ type ClusterRegisterUtil interface {
 	GetCluster(astraHost, cloudId, clusterId, apiToken string) (Cluster, error)
 	CreateCluster(astraHost, cloudId, astraConnectorId, apiToken string) (ClusterInfo, error)
 	UpdateCluster(astraHost, cloudId, clusterId, astraConnectorId, apiToken string) error
-	CreateOrUpdateCluster(astraHost, cloudId, clusterId, astraConnectorId, clustersMethod, apiToken string) (ClusterInfo, error)
+	CreateOrUpdateCluster(astraHost, cloudId, clusterId, astraConnectorId, connectorInstall, clustersMethod, apiToken string) (ClusterInfo, error)
 	GetStorageClass(astraHost, cloudId, clusterId, apiToken string) (string, error)
-	CreateManagedCluster(astraHost, cloudId, clusterID, storageClass, apiToken string) error
-	UpdateManagedCluster(astraHost, clusterId, astraConnectorId, apiToken string) error
-	CreateOrUpdateManagedCluster(astraHost, cloudId, clusterId, astraConnectorId, managedClustersMethod, apiToken string) (ClusterInfo, error)
+	CreateManagedCluster(astraHost, cloudId, clusterID, storageClass, connectorInstall, apiToken string) error
+	UpdateManagedCluster(astraHost, clusterId, astraConnectorId, connectorInstall, apiToken string) error
+	CreateOrUpdateManagedCluster(astraHost, cloudId, clusterId, astraConnectorId, connectorInstall, managedClustersMethod, apiToken string) (ClusterInfo, error)
```

Comment:
Most of the changes here are to take "connectorInstall" into consideration. 
If connectorInstall is set - this means we are in the new arch-3.0 flow using the dummy cluster.

This means we want to skip Getting and Setting the default SC when this is set. Otherwise we want to do things the old way